src/hotkey/key_window.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <stdint.h>
+
+/*
+ * The key_window module provides a FIFO key_window interface to help with bookkeeping the
```

Comment:
Good point on the lack of foreach in ring array.
The tradeoff makes sense for now.